lms/services/lti_h.py

```diff
@@ -23,7 +23,9 @@ def __init__(self, _context, request):
         self._h_user = request.lti_user.h_user
 
         self._authority = request.registry.settings["h_authority"]
-        self._ai_getter = request.find_service(name="ai_getter")
+        self._application_instance_service = request.find_service(
+            name="application_instance"
+        )
```

Comment:
Like the service above, it seems what we really want to know is if provisioning is on, rather than actually keeping the whole application instance around.

Also I kind of hate having the service be it's own factory now we've started doing it the other way...

Reply:
I think it's probably best not to do this because `ApplicationInstanceService.get()` can raise `ConsumerKeyError` so if `LTIHService.__init__()` calls `ApplicationInstanceService.get().provisioning` then `LTIHService.__init__()` can now raise `ConsumerKeyError`, which means that `request.find_service(name="lti_h")` can raise `ConsumerKeyError`. I think it's probably best that `find_service()` doesn't raise, as I think that'll be quite unexpected, easily missed. Even if the `:raise:` were documented in `LTIHService.__init__()` I think it'd be likely missed as you wouldn't think to look for it. Service factories shouldn't raise, which means service `__init__()`'s shouldn't raise unless the factory is going to handle that. It's better for `LTIHService.sync()` to raise `ConsumerKeyError` and document it in its docstrings. _Normal_ methods raising exceptions is something that developers expect and know to look out for.

`LTIHService` having a `self._application_instance_service` might come in handy in future if it needs to use any future `ApplicationInstanceService` methods.